### PR TITLE
wip: add R bindings

### PR DESCRIPTION
**Pull request by joeddav**
This adds R bindings by using _Reticulate_ to interface directly with the Python bindings. The code is functional and can be installed on linux or windows using the `install_github` devtools function. See the README for more information.

Documentation, examples, and tests still a work in progress.

### CODE REVIEW

**Comment by liqunfu** [comment]
Do you need to add this binary file? layers-layers.R.swp

**Comment by KeDengMS** [comment]
Leave a comment

File: bindings/R/R/debugging-profiler.R [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+}
+
+#' @export
+enable_profiler <- function() {
+	cntk$debugging$profiler$enable_profiler()
+}
+
+#' @export
+start_profiler <- function(dir = 'profiler', sync_gpu = TRUE,
+						   reserve_mem = 33554432) {
```
Comment (liqunfu) [inline bindings/R/R/debugging-profiler.R]: Why is number 33554432?

File: bindings/R/R/core.R [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+		read_only = read_only
+	)
+}
+
+#' @export
+value_one_hot <- function(batch, num_classes, device = NULL) {
+	cntk$core$Value$one_hot(
+		to_int(batch),
+		to_int(num_classes),
+		dtype = np$float32,
```
Comment (liqunfu) [inline bindings/R/R/core.R]: In general, we need to support float64 and auto precision too.

File: bindings/R/R/core.R [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+		device = device
+	)
+}
+
+#' @export
+value_asarray <- function(value) {
+	cntk$core$asarray(
+		value,
+		dtype = np$float32
```
Comment (liqunfu) [inline bindings/R/R/core.R]: In general, we need to support float64 and auto precision too.

File: bindings/R/R/io.R [hunk 4]
```diff
@@ -0,0 +1,415 @@
...
+#'
+#' next_minibatch
+#'
+#' restore_mb_from_checkpoint
+#'
+#' mb_stream_infos
+#'
+#' @export
+MinibatchSourceFromData <- function(data_streams,
+									max_samples = 1844674407379551615) {
```
Comment (liqunfu) [inline bindings/R/R/io.R]: In general, use named constants for readability

File: bindings/R/R/layers-blocks.R [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+#' @export
+LSTM <- function(shape, cell_shape = NULL, activation = op_tanh,
+				 use_peepholes = FALSE, init = init_glorot_uniform(),
+				 init_bias = 0, enable_self_stabilization = FALSE, name = '') {
+	cntk$layers$blocks$LSTM(
+		to_int(shape),
+		cell_shape = to_int(cell_shape),
+		activation = activation,
+		use_peepholes = use_peepholes,
+		init = init_glorot_uniform(),
```
Comment (liqunfu) [inline bindings/R/R/layers-blocks.R]: init instead

**Comment by cha-zhang** [comment]
I'm closing this PR because as we agreed R binding will be in a separate repository.

**Pull request closed without merging**